front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionSelector.tsx

```diff
@@ -38,37 +56,78 @@ export default function PowerRestrictionSelector() {
+    const curvesMode = rollingStockToClean.effort_curves.modes;
+    const curvesModesKey = Object.keys(curvesMode);
+
+    let parsedElectrification: ElectrificationPR = {};
+    for (const mode of curvesModesKey) {
+      let powerCodes = curvesMode[mode].curves.map((curve) => curve.cond?.power_restriction_code);
+      powerCodes = powerCodes.filter(
+        (value, index, array) => value && array.indexOf(value) === index
+      );
+      parsedElectrification = {
+        ...parsedElectrification,
+        [mode]: powerCodes,
+      };
+    }
+    return parsedElectrification;
+  };
+
+  const definePowerRestrictionRange = async (powerRestrictionCode?: string) => {
+    if (powerRestrictionCode && pathFinding) {
+      const pathLength = Math.round(
+        lengthFromLineCoordinates(pathFinding.geographic?.coordinates) * 1000
+      );
+      const powerRestrictionRange: PowerRestrictionRange[] = [
+        {
+          begin_position: 0,
+          end_position: pathLength,
+          power_restriction_code: powerRestrictionCode,
+        },
+      ];
+      dispatch(updatePowerRestriction(powerRestrictionRange));
+
+      if (rollingStock && pathWithCatenaries) {
+        const parsedElectrification = cleanConditionalEffortCurves(rollingStock);
+
+        const pathCatenaryRanges = pathWithCatenaries.catenary_ranges;
+        if (pathCatenaryRanges) {
+          // Extract path electrification mode and check compatibility
+          const pathElectrification = pathCatenaryRanges.map((range) => range.mode);
+
+          // Display an error when the first incompatibility is encountered
+          pathElectrification.some((electrification) => {
+            const isInvalid =
+              electrification &&
+              !parsedElectrification[electrification as keyof ElectrificationPR].includes(
+                powerRestrictionCode
+              );
+            if (isInvalid) {
+              dispatch(
+                setFailure({
+                  name: t('errorMessages.error'),
+                  message: t('errorMessages.powerRestrictionInvalidCombination', {
+                    powerRestrictionCode,
+                    electrification,
+                  }),
+                })
+              );
+            }
+
+            return isInvalid;
+          });
+        }
+      }
     } else dispatch(updatePowerRestriction(undefined));
   };
 
   useEffect(() => {
-    if (rollingStockID) {
-      dispatch(updatePowerRestriction(undefined));
-      setPowerRestrictions(undefined);
-      getRollingStockById({ id: rollingStockID })
-        .unwrap()
-        .then((rollingStockInfo) => {
-          getPowerRestrictionsList(rollingStockInfo);
-        });
+    if (rollingStock) {
+      getPowerRestrictionsList(rollingStock);
     }
-  }, [rollingStockID]);
+  }, [isSuccess]);
 
   return powerRestrictions && powerRestrictions.length > 0 && pathFindingID ? (
```

Comment:
If you have modified getPowerRestrictionsList, you need to modify this condition too
```suggestion
  return powerRestrictions.length > 1 && pathFindingID ? (
```

Reply:
Done